app/services/analytics_events.rb

```diff
@@ -1306,9 +1306,113 @@ def idv_doc_auth_submitted_pii_validation(
+  # @option proofing_results [Hash<String,Numeric>] context.stages.state_id.requested_attributes An object whose keys are field names and values are "1" representing PII attributes sent to the state ID proofing vendor for verification.
+  # @option proofing_results [Boolean] context.stages.state_id.timed_out Whether the request to the state ID verification vendor timed out
+  # @option proofing_results [String] context.stages.state_id.transaction_id Vendor-specific transaction ID for the request made to the state id proofing vendor
+  # @option proofing_results [String] context.stages.state_id.vendor_name Name of the vendor used for state ID PII verification. If the ID was not from a supported jurisdiction, it will be "UnsupportedJurisdiction". It MAY also be "UnsupportedJurisdiction" if state ID verification was not needed because other vendor calls did not succeed.
+  # @option proofing_results [String] context.stages.state_id.state The state that was listed as the user's address on their state ID. Note that this may differ from state_id_jurisdiction.
+  # @option proofing_results [String] context.stages.state_id.state_id_jurisdiction The state that issued the drivers license or ID card being used for proofing.
+  # @option proofing_results [String] context.stages.state_id.state_id_number A string describing the _format_ of the state ID number provided.
+  # @option proofing_results [Hash] context.stages.threatmetrix Object holding details about the call made to the device profiling vendor
+  # @option proofing_results [String] context.stages.threatmetrix.client Identifier string indicating which client was used.
+  # @option proofing_results [Boolean] context.stages.threatmetrix.success Whether the request to the vendor succeeded.
+  # @option proofing_results [Hash] context.stages.threatmetrix.errors Hash describing errors encountered when making the request.
+  # @option proofing_results [String,nil] context.stages.threatmetrix.exception If an exception was encountered making the request to the vendor, its message is provided here.
+  # @option proofing_results [Boolean] context.stages.threatmetrix.timed_out Whether the request to the vendor timed out.
+  # @option proofing_results [String] context.stages.threatmetrix.transaction_id Vendor-specific transaction ID for the request.
+  # @option proofing_results [Hash] context.stages.threatmetrix.response_body JSON body of the response returned from the vendor. PII has been redacted.
+  # @option proofing_results [String] context.stages.threatmetrix.response_body.account_lex_id LexID associated with the response.
+  # @option proofing_results [String] context.stages.threatmetrix.response_body.session_id Session ID associated with the response.
+  # @option proofing_results [String] context.stages.threatmetrix.review_status One of "pass", "review", "reject".
```

Comment:
WDYT of trying to update https://github.com/18F/identity-idp/blob/main/spec/support/fake_analytics.rb#L73 to work with nested hashes/analytics like this?

Reply:
I don't think I understand that code enough to do it in this PR. Honestly, it starts to feel to me like we need a typechecker. I kind of wonder if it would be preferable to log  flat, un-nested structures?

Reply:
At that point we should switch to Crystal!